### DISCUSSION TITLE
REST accept non json response

### DISCUSSION THREAD

**Discussion by samg666** [Budibase/budibase#3168] [Help]
Hello,

I am able to use Budibase REST to send email via Integromat webhook successfully, the only issue is the REST always got error "invalid json response body" as Integromate webhook returned plaint text "ACCEPT". I tried to find out if Integromat can return in JSON but seems no such option. So, is there any way BB can accept non json respon?

![iShot2021-10-27 11 48 09](https://user-images.githubusercontent.com/29649651/138996742-c6923a84-87a1-4a3a-a15c-aa221151db71.png)

**Comment by samg666** [comment]
I have find the way to let Integromate response in JSON. But it will be good if BB can accept plaint text as well.

**Reply by shogunpurple** [reply]
If you set your `Accept` header to `text/html`, we should return a string. Does this work for you?

**Reply by samg666** [reply]
@shogunpurple Hi Martin, I tried to set header "Accept: text/html" and seems still not work. As I am not able to custom Slack webhook response in JSON, then Budibase has no way to work with Slack webhook directly. Now I have find a workaround, Budibase POST to Integromate webhook(it will response in JSON), then Integromate POST to Slack.

![image](https://user-images.githubusercontent.com/29649651/139254607-9ca7f027-62a6-4163-be9e-d26b1a9f161d.png)

**Comment by samg666** [comment]
@mjashanks Is there any way Budibase RESET will accept non JSON response? I met the same issue while sending message via Slack webhook, the slack just response plaint text "ok" instead of JSON. then BB raised error as it expecting JSON response.

**Reply by mjashanks** [reply]
Not at the moment, sorry. We are going to do a lot of work on the rest connector in December.
